### ISSUE TITLE
wrap new PLS prior on STIR master

### ISSUE THREAD

**Issue by KrisThielemans** [CCPPETMR/SIRF#126]
@danieldeidda recently added the PLS anatomical prior to STIR. can we wrap this in SIRF? Please put on a separate branch.

**Comment by evgueni-ovtchinnikov** [comment]
wrapped (on SIRF devel), meaningful tests needed

**Comment by KrisThielemans** [comment]
test: use small example, run recon by hand first, check visually, then do automatic test based on ROI (as in others). need same type of test without prior first.

**Comment by evgueni-ovtchinnikov** [comment]
you have lost me here, sorry

i probably lost you too, so trying again: i need a meaningful anatomic image, kappa etc to make sure i made no mistakes when exposing them to sirf

i tried exact image as anatomical, but saw no tangible improvement, which is a bit puzzling to me

**Comment by KrisThielemans** [comment]
I've merged https://github.com/UCL/STIR/pull/255. Note that you'll have to update the version_config.cmake for the SIRF-SuperBuild before you can add that functionalitty to SIRF.

For test case, as we discussed: with noise, the prior should reduce noise. without noise, the prior should smooth more over edges which are not in the anatomical image. It's not a unit-test, but could be in the demo.

**Comment by evgueni-ovtchinnikov** [comment]
building SIRF-SuperBuild failed with CMake complaining about boost being too old (1.58)

i thought SIRF-SuperBuild should download and install the required boost (and other pre-requisits)?

PLSPrior seems to work ok now that i realized what i was doing wrong: i used the exact phantom as anatomical, but it was piecewise-constant, and so PLS prior was zero except on boundaries; using reconstructed image as anatomical made a visible difference

**Comment by KrisThielemans** [comment]
Did you try on Windows? On Windows, we don't attempt to build boost (just too hard to get it to work with shared libs etc). We could possibly downgrade our boost requirement, currently
```
set(Boost_REQUIRED_VERSION 1.64.0)
```
but I think that's a bad idea. Just upgrade your boost version.

Regarding the testing, the PLS prior is not zero in flat regions, it resolves to TV (or actually "smoothed TV" when alpha>0). So using noisy data, it should still have an effect. Indeed, it's sort of the other way around: an anatomical prior is designed to penalise less at anatomical edges (and more in flat regions).

**Comment by evgueni-ovtchinnikov** [comment]
\phi(f) = \sqrt{\alpha^2 + |\nabla v|^2 - {\langle\nabla f,\xi\rangle}^2}
    \xi = \frac{\nabla v}{\sqrt{|\nabla v|^2 + \eta^2}}

so, where `\nabla v = 0, \phi = \alpha`, i.e. does not depend on the image `f` at all?

**Comment by KrisThielemans** [comment]
ah. typo! corrected https://github.com/UCL/STIR/commit/4a75f6b97c0ce38ebada5f99f6732bbe5ad5cb3d
```
\phi(f) = \sqrt{\alpha^2 + |\nabla f|^2 - {\langle\nabla f,\xi\rangle}^2}
```
thanks for spotting this!

**Comment by evgueni-ovtchinnikov** [comment]
closed by 72c4494563bc96a78c5413ff3a5da48fd910048e
